specifyweb/frontend/js_src/lib/components/Merging/index.tsx

```diff
@@ -251,6 +257,51 @@ function Merging({
   );
 }
 
+function MergeButton<SCHEMA extends AnySchema>({
+  id,
+  mergeResource,
+}: {
+  readonly id: (suffix: string) => string;
+  readonly mergeResource: SpecifyResource<SCHEMA>;
+}): JSX.Element {
+  const [formId] = React.useState(id('form'));
+  const [saveBlocked, setSaveBlocked] = React.useState(false);
+  const [showSaveBlockedDialog, setShowBlockedDialog] = React.useState(false);
+
+  React.useEffect(() => {
```

Comment:
We already have a hook `useSaveBlockers`. Can you check if that works for you here?
If not, can you modify that hook to work for you?

Also check the other place where you added a blockers change listener hook to see if you can replace it with `useSaveBlockers`